src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2865,6 +2962,18 @@ public bool Contains(TKey key)
                 } while (g != _lastGrouping);
             }
         }
+        
+        public IGrouping<TKey, TElement>[] ToArray()
+        {
+            IGrouping<TKey, TElement>[] array = new IGrouping<TKey, TElement>[_count];
+            int index = 0;
+            foreach(IGrouping<TKey, TElement> group in this)
```

Comment:
I wonder if it'd be better to manually loop through the groupings here, instead of `foreach`, to avoid the enumerator allocation.